src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -118,7 +118,8 @@ private RecommendedSymbols GetSymbolsOffOfContainer()
                 RangeExpressionSyntax rangeExpression => GetSymbolsOffOfExpression(rangeExpression.LeftOperand),
                 QualifiedNameSyntax qualifiedName => GetSymbolsOffOfName(qualifiedName.Left),
                 AliasQualifiedNameSyntax aliasName => GetSymbolsOffOffAlias(aliasName.Alias),
-                MemberBindingExpressionSyntax _ => GetSymbolsOffOfConditionalReceiver(node.GetParentConditionalAccessExpression()!.Expression),
+                MemberBindingExpressionSyntax _ when node.GetParentConditionalAccessExpression() is ConditionalAccessExpressionSyntax conditionalAccessExpression
+                    => GetSymbolsOffOfConditionalReceiver(conditionalAccessExpression.Expression),
```

Comment:
i'm a little confused.  If we had a MemberBindingExpression, how did we not get a parent conditional access expression?  that shoudl always succeed.  Are we missing a case in GetParentConditionalAccessExpression()?

Reply:
ah, it looks like we didn't update GetParentConditionalAccessExpression when this change was made:

https://github.com/dotnet/roslyn/blob/724400b4a684aa58ee520f18183cc15089796205/src/Compilers/CSharp/Portable/Parser/LanguageParser.cs#L11209-L11217

Reply:
Thanks for pointing this out. I changed the code to handle optional `!`s instead